manifest_indexing/utils.py

```diff
@@ -152,3 +169,35 @@ def upload_file_to_s3_and_generate_presigned_url(
             return presigned_url
 
     return None
+
+
+def check_user_permission(access_token):
+    """
+    Check if user has permission to run the job or not
+
+    Args:
+        access_token(str): the access token
+    Returns:
+        bool: if user has permission to run the job or not
+        dict: a message log
+    """
+    response = requests.get(
+        "{}/auth/mapping".format(ARBORIST_URL), headers={"Authorization": access_token}
```

Comment:
instead of getting the full auth mapping you could make an auth request to arborist with the access you want to check. 2 different ways of doing the same thing, I personally think it might be clearer/more efficient to do the latter: http://petstore.swagger.io/?url=https://raw.githubusercontent.com/uc-cdis/arborist/master/docs/openapi.yaml#/auth/post_auth_request . So the flow is: "can this user access jobs and can this user write in indexd"? instead of "give me all the user's information and I will make a decision". The first flow puts the burden of the decision on Arborist which I think is a better pattern to follow